From a7cc0e6bee51f284d9eddf34cd513534c0225710 Mon Sep 17 00:00:00 2001
From: Jon Lin <jon.lin@rock-chips.com>
Date: Sat, 6 Feb 2021 09:45:45 +0800
Subject: [PATCH] drivers: rkflash: Remove cache bitflip detect 2

Part of spinand sram maybe change after read status register

Fixes: cf69491c97cc ("drivers: rkflash: Add spinand program cache recheck")
Change-Id: Ia8f902fe51562d71a5b8e78a80e63eb26257df38
Signed-off-by: Jon Lin <jon.lin@rock-chips.com>
---
 drivers/rkflash/sfc_nand.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/drivers/rkflash/sfc_nand.c b/drivers/rkflash/sfc_nand.c
index 7cf787d65fb8..612ab2545be7 100644
--- a/drivers/rkflash/sfc_nand.c
+++ b/drivers/rkflash/sfc_nand.c
@@ -670,18 +670,6 @@ u32 sfc_nand_prog_page_raw(u8 cs, u32 addr, u32 *p_page_buf)
 		return ret;
 
 	ret = sfc_nand_wait_busy(&status, 1000 * 1000);
-	/*
-	 * At the moment of power lost, flash maybe work in a unkonw state
-	 * and result in bit flip, when this situation is detected by cache
-	 * recheck, it's better to wait a second for a reliable hardware
-	 * environment to reduce error behavior.
-	 */
-	sfc_nand_read_cache(addr, (u32 *)sfc_nand_dev.recheck_buffer, 0, data_area_size);
-	if (memcmp(sfc_nand_dev.recheck_buffer, p_page_buf, data_area_size)) {
-		rkflash_print_error("%s cache bitflip2\n", __func__);
-		msleep(1000);
-	}
-
 	if (status & (1 << 3))
 		return SFC_NAND_PROG_ERASE_ERROR;
 
-- 
2.35.3

